File: Arrays/increasing-triplet-subsequence/increasing-triplet-subsequence.go
// https://leetcode.com/problems/increasing-triplet-subsequence/
package main

import "math"
func increasingTriplet(nums []int) bool {
    min := math.MaxInt32
    min2 := math.MaxInt32

    for _, num := range nums {
        if num <= min {
            min = num
        } else if num <= min2 {
            min2 = num
        } else {
            return true
        }
    }

    return false
}

File: Arrays/increasing-triplet-subsequence/increasing-triplet-subsequence.ts
// https://leetcode.com/problems/increasing-triplet-subsequence/
function increasingTriplet(nums: number[]): boolean {
  let min = Infinity;
  let min2 = Infinity;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] <= min) {
      min = nums[i];
    } else if (nums[i] <= min2) {
      min2 = nums[i];
    } else {
      return true;
    }
  }

  return false;
}
